engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -68,6 +68,10 @@ public static Type create(
     return result;
   }
 
+  public static Type noType() {
```

Comment:
What's this? You need a `null`-like key into `ConcurrentHashMap`! But such a key should not be publicly visible - it should be private to the usage class.